### PR TITLE
HTML Template support Issue #513 Master

### PR DESCRIPTION
**Pull request by zangsir**
modified antlr4 grammar and related methods to recognize HTML template in the config file for HTMLVis (those that contains %%value%% and %%anno%%). Modified SpanHTMLOutputter to handle the html template case: replace the %%value%% with the actual value or replace %%anno%% with the actual annotation. The HTML template visualizer will enable clicking on the links and it will take you directly to an external website (such as an entry for the current token in dictionary.com).

### CODE REVIEW

**Comment by thomaskrause** [comment]
Can you resubmit this pull request against the korpling:develop branch instead of korpling:master?

File: annis-visualizers/src/main/antlr4/annis/visualizers/htmlvis/HTMLVisConfig.g4 [hunk 1]
```diff
@@ -34,18 +34,21 @@ NEWLINE : '\n';
...
+temp: TEMPVALUE|TEMPANNO;
+innertype: ~(QUOTE|TEMPVALUE|TEMPANNO)+;
+innerhtmltemp: (innertype* temp innertype*)+; 
 innermeta: ~(QUOTE|WS|NEWLINE)+;
 
 type
   : VALUE # typeValue
   | ESCAPED_VALUE # typeEscapedValue
   | ANNO # typeAnno
+  | QUOTE innerhtmltemp QUOTE # typeHtmlTemp
```
Comment (thomaskrause) [inline annis-visualizers/src/main/antlr4/annis/visualizers/htmlvis/HTMLVisConfig.g4]: I can't see how the parser can distinguish between the "innertype" and "innerhtmltemp" case. Both start and end with quotes and innertype even allows explicitly to have the TEMPVALUE and TEMPANNO token.

Comment (zangsir) [inline annis-visualizers/src/main/antlr4/annis/visualizers/htmlvis/HTMLVisConfig.g4]: actually I thought the "~" in " innertype: ~(QUOTE|TEMPVALUE|TEMPANNO)+  " means that innertype doesn't allow QUOTE or TEMPVALUE or TEMPANNO, no?

**Comment by thomaskrause** [comment]
moved to #522

**Pull request closed without merging**